src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpApplicationPreambleController.cs

```diff
@@ -712,28 +727,14 @@ public async Task<IActionResult> ChangeProviderRoute(ChangeProviderRouteViewMode
             var applicationDetails = _sessionService.Get<ApplicationDetails>(ApplicationDetailsKey);
             if (!ModelState.IsValid)
             {
-                var providerRoutes = await _roatpApiClient.GetApplicationRoutes();
-
-                var existingProviderRoute = providerRoutes.FirstOrDefault(x => x.Id == applicationDetails.RoatpRegisterStatus.ProviderTypeId);
+                var modelErrors = ModelState.Values.SelectMany(v => v.Errors).FirstOrDefault();
 
-                model = new ChangeProviderRouteViewModel
+                if (modelErrors?.ErrorMessage!=null)
                 {
-                    UKPRN = applicationDetails.UKPRN.ToString(),
-                    CurrentProviderType = existingProviderRoute,
-                    ErrorMessages = new List<ValidationErrorDetail>()
-                };
-
-                var modelErrors = ModelState.Values.SelectMany(v => v.Errors);
-                foreach (var modelError in modelErrors)
-                {
-                    model.ErrorMessages.Add(new ValidationErrorDetail
-                    {
-                        Field = "ChangeApplicationRoute",
-                        ErrorMessage = modelError.ErrorMessage
-                    });
+                    TempData[_employerProviderValidationMessage] = modelErrors.ErrorMessage;
```

Comment:
We should be returning view here as the model state is invalid, isn't it?

Reply:
that's what causes the post to get put into the browser history, so we are avoiding that in all the PRG stories.

Reply:
It follows the pattern established in APR-2893 of writing the message to tempData and redirecting back to the GET